src/FFmpegReader.cpp

```diff
@@ -1352,7 +1352,13 @@ void FFmpegReader::ProcessVideoPacket(int64_t requested_frame) {
 		std::shared_ptr<Frame> f = CreateFrame(current_frame);
 
 		// Add Image data to frame
-		f->AddImage(width, height, 4, QImage::Format_RGBA8888_Premultiplied, buffer);
+		if (!ffmpeg_has_alpha(AV_GET_CODEC_PIXEL_FORMAT(pStream, pCodecCtx))) {
+			// Add image with no alpha channel, Speed optimization
+			f->AddImage(width, height, 4, QImage::Format_RGBA8888_Premultiplied, buffer);
+		} else {
+			// Add image with alpha channel (this will be converted to premultipled when needed, but is slower)
+			f->AddImage(width, height, 4, QImage::Format_RGBA8888, buffer);
```

Comment:
Hmm. It'd be good to get a test file with alpha channel, if possible, to unit-test this. Presumably someone must have one, if they noticed this in the first place?

Reply:
I've recently learned that there are many formats which can allow an alpha channel, and thus, we can always add a transparent video into our unittests (I like that idea), but it won't test for all possible alpha formats. So, it still leaves the possibility that a format is not working correctly, but I suppose that will just be the case.